stockflux-securities-master/src/services/SecuritiesService.js

```diff
@@ -20,3 +20,28 @@ export async function getSecurity(securityId) {
   const security = await response.json();
   return security;
 }
+
+export async function postSecurity(security) {
+  const options = await getWindowOptions();
+  const response = await fetch(
+    `${options.customData.apiBaseUrl}/securities-v2`,
+    {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json"
+      },
+      body: JSON.stringify(security)
+    }
+  );
+  if (response.status === 201) {
+    const responseJSON = await response.json();
+    return responseJSON;
+  } else if (response.status === 400) {
+    const responseJSON = await response.json();
+    throw new Error(responseJSON.messages);
```

Comment:
You should be careful here. Usually, you would pass a string message into the `Error` object but here you're passing an array of messages. When it comes to handling the error in the `.catch(...)`, then you will need to check the type of the errors and treat them differently. If you take a look at how the current `StockFluxService` is implemented, things seem to be done a little differently. I would try to stick to the existing convention where possible.

Reply:
I've moved this logic into the form and made the service return an object with a status code and a message. What do you think about this approach?